### DISCUSSION TITLE
[react-instantsearch-hooks] `useMenu` + `useRefinementList` on a single facet

### DISCUSSION THREAD

**Discussion by avremel** [algolia/instantsearch#5347] [Q&A] [answered]
For a mobile e-commerce UI, I would like to display a single facet (eg, Brand) both as a `RefinementList` and as a `Menu`.

When I applied both widgets to the same facet, Algolia returns 0 `items` for both widgets, presumably because these two widgets can't coexist: If I select two brands in a `RefinementList`, what would I display as selected in the `Menu`? And if I make a `Menu` selection, do I ignore the  multiple `RefinementList` selections?

Although this all makes sense, I was wondering if I can nonetheless bypass this constraint and display both widgets. The functionality would behave roughly like so: If a user makes multiple `RefinementList` selections, they get applied and any prior `Menu` selection is disregarded. If the user makes a `Menu` selection,  only that selection is applied, and any prior `RefinementList` selections are disregarded.

Are there any primitives which allow for this behavior?

About `useMenu` specifically:
- Is it possible to display an exhaustive list of every facet (without having to toggle a show more button).
- If I can't get an exhaustive list, can I at least display a search bar (like `useRefinementList`)?

**Comment by Haroenv** [comment, marked as answer]
Two different widgets can't coexist with different behaviour unfortunately, because we'd either need to make the refinementList act like a menu (remove a different refinement if it's already set) or menu as a refinement list (having multiple items checked).

To be able to have the best of both worlds, you likely want to use only useRefinementList, and to set only a single value you can do `useInstantSearch().setIndexUiState(state => ({ ...state, refinementList: {...state.refinementlist, [attribute]: [selectedValue] }}))`
